### PR TITLE
Improve the animation example a little

### PR DESCRIPTION
**Pull request by Hixie**
Apply some of the review comments from #1.

### CODE REVIEW

File: lib/common.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+// Suppose this comes from a package and cannot be known by any
+// libraries we invent (and doesn't know about our libraries).
+mixin ShowAnimationsMixin<T extends StatefulWidget> on State<T> {
+  bool get showAnimations => _showAnimations;
+  bool _showAnimations = true;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
```
Comment (TimWhiting) [inline lib/common.dart:11]: Problem is that the mixin's definition of didChangeDependencies replaces the StatefulAnimationState's & _ExampleState's definition. So you can't separate the logic out like this using stateful widgets. This is one of the issues that is addressed by hooks.

File: lib/common.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    _showAnimations = !MediaQuery.of(context).disableAnimations;
```
Comment (rrousselGit) [inline lib/common.dart:12]: Why use MediaQuery here but in the widget use TickerMode (and ignore the TickerMode.enabled flag)?

Comment (rrousselGit) [reply lib/common.dart:12]: Forget about it, it's my mistake

File: lib/common.dart (hunk 1 above)
Comment (Hixie) [reply lib/common.dart:11]: I'm not sure I follow what you mean by "replaces". It's just inheritance, they all call each other, no?

Comment (TimWhiting) [reply lib/common.dart:11]: Forgive me. In this case it is just inheritance. I was thinking of multiple mixins and the problem where one mixin's definition would shadow the other mixin's definition.

**Pull request merged**